File: frontend/src/lib/services/petfriendly.ts
import { apiFetch } from "../api";

export interface PetfriendlyPlace {
    id: string;
    added_by: string;
    name: string;
    category: string;
    address: string | null;
    city: string | null;
    description: string | null;
    image_url: string | null;
    latitude: number | null;
    longitude: number | null;
    phone: string | null;
    website: string | null;
    rating: number;
    pet_sizes_allowed: string;
    has_water_bowls: string;
    has_pet_menu: string;
    created_at: string | null;
}

export interface PlaceCreate {
    name: string;
    category: string;
    address?: string;
    city?: string;
    description?: string;
    image_url?: string;
    latitude?: number;
    longitude?: number;
    phone?: string;
    website?: string;
    rating?: number;
    pet_sizes_allowed?: string;
    has_water_bowls?: string;
    has_pet_menu?: string;
}

export async function getPlaces(category?: string, city?: string): Promise<PetfriendlyPlace[]> {
    const params = new URLSearchParams();
    if (category) params.append("category", category);
    if (city) params.append("city", city);
    const qs = params.toString() ? `?${params.toString()}` : "";
    return apiFetch<PetfriendlyPlace[]>("perdidas", `/places/${qs}`);
}

export async function createPlace(place: PlaceCreate): Promise<PetfriendlyPlace> {
    return apiFetch<PetfriendlyPlace>("perdidas", "/places/", {
        method: "POST",
        body: JSON.stringify(place),
    });
}

File: backend/michicondrias_perdidas/app/models/petfriendly.py
import uuid
from sqlalchemy import Column, String, Text, DateTime, Float, Integer
from sqlalchemy.sql import func
from app.db.session import Base


class PetfriendlyPlace(Base):
    __tablename__ = "petfriendly_places"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    added_by = Column(String, nullable=False, index=True)

    name = Column(String(200), nullable=False)
    category = Column(String(60), nullable=False)  # restaurante, cafe, parque, hotel, tienda, veterinaria
    address = Column(String(300), nullable=True)
    city = Column(String(100), nullable=True)
    description = Column(Text, nullable=True)
    image_url = Column(Text, nullable=True)

    latitude = Column(Float, nullable=True)
    longitude = Column(Float, nullable=True)

    phone = Column(String(30), nullable=True)
    website = Column(String(255), nullable=True)
    rating = Column(Integer, default=0)  # 1-5

    pet_sizes_allowed = Column(String(60), default="todos")  # pequeños, medianos, grandes, todos
    has_water_bowls = Column(String(10), default="no")
    has_pet_menu = Column(String(10), default="no")

    created_at = Column(DateTime(timezone=True), server_default=func.now())
